conda/petsc/meta.yaml

```diff
@@ -1,7 +1,7 @@
 # Do not use jinja templating (A physical change to this file is required to trigger a build)
-{% set build = 7 %}
+{% set build = 0 %}
 {% set strbuild = "build_" + build|string %}
-{% set version = "3.15.1" %}
+{% set version = "3.16.5" %}
```

Comment:
Don't we want to set

```
  script_env:
    - MOOSE_JOBS
```

in this too?

Reply:
PETSc ./configure does not understand this variable. @fdkong as I recall, we did _something_ to make it understand how many cores we wished to use (so we were not oversubscribing on HPC head nodes).

Reply:
We need to add this to the configure: https://github.com/idaholab/moose/blob/a5afafb36f97de487f062f94d61ed4903a465b43/scripts/update_and_rebuild_petsc.sh#L103

Reply:
We used to have this. I am not sure how that was removed ..

Reply:
Never mind. Still there

Reply:
Moved the `MAKE_NP_STR` logic into `configure_petsc.sh` so that the conda and the update script can both use it. Conda now pulls `MOOSE_JOBS` from the build environment if it exists and uses it for both the libmesh and petsc builds.